[lint-diff] No callstack after crash

LintDiff had a crash bug, and the code bug itself is being tracked by #38492.  A secondary bug is that no callstack was printed after the crash:

```
Wed, 29 Oct 2025 17:07:35 GMT Correlating runs...
Wed, 29 Oct 2025 17:07:35 GMT Error: Process completed with exit code 1.
```

Run: https://github.com/Azure/azure-rest-api-specs-pr/actions/runs/18916089095/job/53999750310#step:7:7914
PR: https://github.com/Azure/azure-rest-api-specs-pr/pull/25212

However, when I run the same repro locally, I do get a callstack:

```
stderr: ''
::endgroup::
        Autorest result length: 187720
 
 
Correlating runs...
file:///home/mharder/specs/eng/tools/lint-diff/dist/src/correlateResults.js:58
                throw new Error(`Multiple before candidates found for key ${key} using readme ${readme}`);
                      ^
 
Error: Multiple before candidates found for key specification/newrelic/resource-manager/readme.md#package-2025-05-01-preview using readme Readme(/home/mharder/specs-pr/specification/newrelic/resource-manager/readme.md, {logger: undefined})
    at correlateRuns (file:///home/mharder/specs/eng/tools/lint-diff/dist/src/correlateResults.js:58:23)
    at async runLintDiff (file:///home/mharder/specs/eng/tools/lint-diff/dist/src/lint-diff.js:118:29)
    at async main (file:///home/mharder/specs/eng/tools/lint-diff/dist/src/lint-diff.js:73:5)
    at async file:///home/mharder/specs/eng/tools/lint-diff/cmd/lint-diff.js:5:1
 
Node.js v24.11.0
```

Why didn't this callstack appear in the CI log?

--- a/eng/tools/lint-diff/src/lint-diff.ts
+++ b/eng/tools/lint-diff/src/lint-diff.ts
@@ -157,7 +157,7 @@ async function runLintDiff(
     })
     .filter((result) => result.errors.length > 0);
   if (autoRestErrors.length > 0) {
-    generateAutoRestErrorReport(autoRestErrors, outFile);
+    await generateAutoRestErrorReport(autoRestErrors, outFile);
     console.log("AutoRest errors found. See workflow summary for details.");
 
     process.exitCode = 1;

